### PR TITLE
Refactor Makefile to build a unique XPI per arch

### PR DESCRIPTION
**Pull request by willdurand**
This is an alternative approach to sign the extension as a **regular** add-on.

The workflow I am envisioning to release this extension would be:

1. update `VERSION` in the `Makefile` 
2. run `make package`
3. upload the 4 _unsigned_ XPIs on AMO (as unlisted versions). Note that each XPI has a slightly different version (`x.y.z.0` for `linux`, `x.y.z.1` for `linux64`, `x.y.z.2` for `mac64`, and `x.y.z.3` for `win32`)
4. File a jira ticket to let SREs upload the XPI files and `update.json` files to the FTP

### CODE REVIEW

File: extension/template-manifest.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
 {
   "manifest_version": 2,
   "name": "Firefox DevTools ADB Extension",
-  "author": "Mozilla & Android Open Source Project",
   "version": "@@VERSION@@",
-  "description": "An extension providing ADB binaries for Android remote debugging in Firefox DevTools. May be automatically installed when using about:debugging.",
-  "applications": {
+  "description": "An extension providing (@@ARCH@@) ADB binaries for Android remote debugging in Firefox DevTools. May be automatically installed when using about:debugging.",
+  "browser_specific_settings": {
```
Comment (willdurand) [inline extension/template-manifest.json:6]: `applications` is deprecated.

File: extension/template-manifest.json [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
     "gecko": {
       "id": "adb@mozilla.org",
       "strict_min_version": "64.0a1",
       "update_url": "@@UPDATE_URL@@"
     }
   },
-  "permissions": [
-  ],
-  "homepage_url": "https://github.com/mozilla/devtools-adb-extension"
+  "developer": {
```
Comment (willdurand) [inline extension/template-manifest.json:13]: `homepage_url`/`author` are kinda deprecated.

File: Makefile [hunk 3]
```diff
@@ -1,60 +1,44 @@
...
-		echo "[release-$$arch] Sign .xpi"; \
-		../sign.sh $(XPI_NAME)-$$arch.xpi; \
-		echo "[release-$$arch] Upload .xpi"; \
-		aws s3 cp \
-			$(XPI_NAME)-$$arch.xpi \
-			$(S3_BASE_URL)/$$arch/$(XPI_NAME)-$$arch.xpi; \
-		echo "[release-$$arch] Copy to 'latest' .xpi"; \
-		aws s3 cp \
-			$(S3_BASE_URL)/$$arch/$(XPI_NAME)-$$arch.xpi \
-			$(S3_BASE_URL)/$$arch/adb-extension-latest-$$arch.xpi; \
```
Comment (juliandescottes) [inline Makefile:44]: Currently we rely on having the "latest" version of the xpi always at the same URL, and that's what Firefox will try to download by default:

https://searchfox.org/mozilla-central/source/modules/libpref/init/all.js#3969

Do you think SREs can both move the xpi to the FTP and create a copy with the expected name ? 

Or should we work around that, eg retrieve & read the update.json file to get the update_link?

**Review by juliandescottes — COMMENTED** [review]
Thanks! One small question about the "latest" xpis we currently generate.

File: Makefile (hunk 3 above)
Comment (willdurand) [reply Makefile:44]: I think that means we'll have to prepare all the signed XPIs for SREs, which means 8 XPIs: the 4 signed by AMO, and a copy of these 4 but renamed to "-latest". Does that make sense?

Comment (juliandescottes) [reply Makefile:44]: Yes that sounds good. Let's update the README to mention this extra step?

**Review by juliandescottes — CHANGES_REQUESTED** [review]
Requesting changes to mention copying the signed xpis to "latest" before uploading them to the FTP.

**Review by juliandescottes — APPROVED** [review]

**Pull request merged**